http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeServiceDbQueries.scala

```diff
@@ -264,6 +264,236 @@ object RuntimeServiceDbQueries {
     listRuntimesHelper(labelMap, excludeStatuses, creatorOnly, workspaceId, cp)
   }
 
+  /**
+   * List runtimes filtered by the given terms. Only return authorized resources (per reader*Ids and/or owner*Ids).
+   * @param labelMap
+   * @param excludeStatuses
+   * @param creatorOnly
+   * @param workspaceId
+   * @param cloudProvider
+   * @param readerRuntimeIds
+   * @param readerWorkspaceIds
+   * @param ownerWorkspaceIds
+   * @param readerGoogleProjectIds
+   * @param ownerGoogleProjectIds
+   * @return
+   */
+  def listAuthorizedRuntimes(
+    // Filters
+    labelMap: LabelMap,
+    excludeStatuses: List[RuntimeStatus],
+    creatorOnly: Option[WorkbenchEmail],
+    workspaceId: Option[WorkspaceId],
+    cloudProvider: Option[CloudProvider],
+
+    // Authorizations
+    readerRuntimeIds: List[String],
+    readerWorkspaceIds: List[String],
+    ownerWorkspaceIds: List[String],
+    readerGoogleProjectIds: List[String],
+    ownerGoogleProjectIds: List[String]
+  ): DBIO[Vector[ListRuntimeResponse2]] = {
+    val cp = cloudProvider.map(cp => Right(cp))
```

Comment:
strong agree, that would be imo part of a larger sanifying rewrite of the data model for supporting multiple cloud providers

Reply:
I created the broad task https://broadworkbench.atlassian.net/browse/IA-4609

Reply:
I think we generally use an ADT for CloudContext / CloudProvider: https://github.com/DataBiosphere/leonardo/blob/develop/core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/models.scala#L13-L44

Idiomatically, Either is usually used for error handling (with Left being error, Right being the result) -- sort of like an Option enriched with a failure message.

Generally I would advocate using an ADT to represent a sum type instead of Either.